zeppelin-jupyter/src/main/java/org/apache/zeppelin/jupyter/JupyterUtil.java

```diff
@@ -167,6 +164,100 @@ private Gson getGson(GsonBuilder gsonBuilder) {
         .registerTypeAdapterFactory(outputTypeFactory).create();
   }
 
+  public String getJson(String input) {
+    Note note = getNote(new StringReader(input), "%spark", "%md");
+    return new Gson().toJson(note);
+  }
+
+  public String getNbformat(String note) {
+    Note noteFormat = getGson(new GsonBuilder()).fromJson(note, Note.class);
+
+    JsonObject nbformat = new JsonObject();
+    JsonArray cells = new JsonArray();
+
+    RegularExpression MD = new RegularExpression("%md\\s");
+    RegularExpression SQL = new RegularExpression("%sql\\s");
+    RegularExpression UNKNOWN_MAGIC = new RegularExpression("%\\w+\\s");
+    RegularExpression HTML = new RegularExpression("%html\\s");
+    RegularExpression SPARK = new RegularExpression("%spark\\s");
```

Comment:
hmm, I think perhaps we need more flexibility here - for example the group/interpreter name for Spark could be changed. Same for all interpreter names like md for example.

Reply:
Don't know what you mean here exactly. Do you want me to include conversions for other interpreters like python and sh?

Reply:
I mean in Zeppelin I could have multiple "names" for the Spark interpreter, it might not be `%spark`

Reply:
How can I access the potentially changed name of the interpreter here?

Reply:
would be great to merge this feature so bumping this question :) @ShamsUlAzeem @felixcheung

Reply:
Sorry about the delay. Each interpreter can have multiple instances and in more than one groups and/or have more than one names. (see config example [here](https://github.com/apache/zeppelin/blob/fe07e5a4962f9457fc2aca473511a50a553499b0/spark/interpreter/src/main/resources/interpreter-setting.json#L171) `group` and `name`fields)

One approach is to:
1. get the binded interpreter instead of parsing the text (https://github.com/apache/zeppelin/blob/master/zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java#L241)

2. from the interpreter instance, get the classname (https://github.com/apache/zeppelin/blob/master/zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/Interpreter.java#L240)

3. check the classname against the well-known built-in ones like `org.apache.zeppelin.spark.PySparkInterpreter` (https://github.com/apache/zeppelin/blob/fe07e5a4962f9457fc2aca473511a50a553499b0/spark/interpreter/src/main/resources/interpreter-setting.json#L171)